src/tools/illink/src/ILLink.RoslynAnalyzer/RequiresAssemblyFilesAnalyzer.cs

```diff
@@ -58,6 +61,21 @@ internal override bool IsAnalyzerEnabled (AnalyzerOptions options)
 			return true;
 		}
 
+		internal override bool IsRequiresCheck (Compilation compilation, IPropertySymbol propertySymbol)
+		{
+			// "IsAssemblyFilesSupported" is treated as a requires check for testing purposes only, and
+			// is not officially-supported product behavior.
+			var runtimeFeaturesType = compilation.GetTypeByMetadataName ("ILLink.RoslynAnalyzer.TestFeatures");
+			if (runtimeFeaturesType == null)
+				return false;
+
+			var isDynamicCodeSupportedProperty = runtimeFeaturesType.GetMembers ("IsAssemblyFilesSupported").OfType<IPropertySymbol> ().FirstOrDefault ();
```

Comment:
Did we add code/tests for IsDynamicCodeCompiled? See https://github.com/dotnet/linker/issues/2715#issuecomment-1212577966